src/core/thread/mle.cpp

```diff
@@ -1061,7 +1061,9 @@ otError Mle::AppendLeaderData(Message &aMessage)
 
 void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
 {
-    uint8_t length;
+    uint8_t length = 255; // sizeof( NetworkDataTlv::mNetworkData )
```

Comment:
I agree with changing `mle_tlvs.h` as well, but I also agree that it will cause `#include` issues.

To minimize change in this PR, we could update this line to:
```
   uint8_t length = sizeof(aTlv) - sizeof(Tlv);
```

Thoughts?